R/yyww-to-yyyymmdd.R

```diff
@@ -0,0 +1,46 @@
+#' Convert date format YYWW to YYYY-MM-DD
+#'
+#' Since the exact date isn't given in the input, this function will set the
+#' date to Monday of the week. As a precaution, a leading zero is added if it
+#' has been removed. This can e.g., happen if the input was "0107" and has been
+#' converted to a numeric 107.
+#'
+#' @param yyww Character(s) of the format YYWW.
+#'
+#' @returns Date(s) in the format YYYY-MM-DD.
+#'
+#' @keywords internal
+#'
+#' @examples
+#' yyww_to_yyyymmdd("0101")
+#' yyww_to_yyyymmdd("2402")
+#' yyww_to_yyyymmdd("2453")
+#' yyww_to_yyyymmdd(c("0102", "0304"))
+#' yyww_to_yyyymmdd(c("0100", "0353"))
+#' yyww_to_yyyymmdd(c("0953", "0701"))
+#' yyww_to_yyyymmdd(953)
+yyww_to_yyyymmdd <- function(yyww) {
+  if (is.numeric(yyww)) {
+    # Ensure input is zero-padded to length 4
+    yyww <- sprintf("%04d", yyww)
+  }
+
+  # Extract year and week
+  year <- stringr::str_sub(yyww, 1, 2)
+  week <- stringr::str_sub(yyww, 3, 4)
+
+  # Calculate the first day of the ISO year, which is when the first week
+  # has most of the week days in it (4th of January, or the first Thursday).
+  # See: https://en.wikipedia.org/wiki/ISO_week_date
```

Comment:
Added some clarification comments, since it confused me what the code was actually doing.